ExtraDry/ExtraDry.Blazor/Components/Dry/Forms/DryInputDateTime.razor.cs

```diff
@@ -0,0 +1,113 @@
+﻿using Newtonsoft.Json.Linq;
+
+namespace ExtraDry.Blazor.Forms;
+
+public partial class DryInputDateTime<T> : ComponentBase, IDryInput<T>, IExtraDryComponent
+{
+
+    /// <inheritdoc />
+    [Parameter]
+    public string CssClass { get; set; } = string.Empty;
+
+    /// <inheritdoc />
+    [Parameter, EditorRequired]
+    public T? Model { get; set; }
+
+    /// <inheritdoc />
+    [Parameter, EditorRequired]
+    public PropertyDescription? Property { get; set; }
+
+    /// <inheritdoc />
+    [Parameter]
+    public EventCallback<ChangeEventArgs>? OnChange { get; set; }
+
+    /// <inheritdoc cref="Blazor.EditMode" />
+    [CascadingParameter]
+    public EditMode EditMode { get; set; } = EditMode.Create;
+
+    /// <inheritdoc />
+    [Parameter(CaptureUnmatchedValues = true)]
+    public Dictionary<string, object>? UnmatchedAttributes { get; set; }
+
+    /// <inheritdoc/>
+    [Parameter]
+    public bool ReadOnly { get; set; }
+
+    protected override void OnParametersSet()
+    {
+        if(Model == null || Property == null) {
+            return;
+        }
+
+        Value = DisplayMode switch {
+            "date" => StringValue("yyyy-MM-dd"),
+            "time" => StringValue("HH:mm"),
+            _ => StringValue("yyyy-MM-ddTHH:mm")
+        };
+    }
+
+    private string StringValue(string format)
+    {
+        var prop = Property?.GetValue(Model);
+        if(Property == null || prop == null) {
+            return string.Empty;
+        }
+        if(Property.PropertyType == typeof(DateTime)) {
+            return ((DateTime)prop).ToLocalTime().ToString(format, CultureInfo.InvariantCulture);
+        }
+        else if(Property.PropertyType == typeof(DateOnly)) {
+            return ((DateOnly)prop).ToString(format, CultureInfo.InvariantCulture);
+        }
+        else if(Property.PropertyType == typeof(TimeOnly)) {
+            return ((TimeOnly)prop).ToString(format, CultureInfo.InvariantCulture);
+        }
+        return prop?.ToString() ?? string.Empty;
+    }
+
+    private async Task HandleChange(ChangeEventArgs args)
+    {
+        if(Property == null || Model == null) {
+            return;
+        }
+        var value = args.Value;
+        if(Property.PropertyType == typeof(DateTime)) {
```

Comment:
Swallowing a lot of formatting issues without feedback.  Does this run the risk of having the underlying value not the same as the displayed value?

Reply:
It does, I just edited the html to allow me to make the input invalid, I'm trying to find a precedent for validation messages from inputs to follow in ED and failing that, i'll see what microsoft have to say on the matter

Reply:
DryInput has a valid property and validation message which are displayed, but none of the controls that it renders have a way to populate this, I'll look at adding something to pass in that can be fed back.

Reply:
I've passed in a function that can leverage the validation mechanism that already exists on the DryInput